### PR TITLE
Add codebase utils

### PR DESCRIPTION
**Pull request by chenhao94**
Test: tested with https://github.com/cyfitech/cris-core/pull/12

### CODE REVIEW

File: scripts/codebase_utils.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+for cmd in bazel; do
+    which "$cmd" >/dev/null
+done
+
```
Comment (xkszltl) [inline scripts/codebase_utils.sh:8]: Are we even building anything without that DIRS env var?

Comment (chenhao94) [reply scripts/codebase_utils.sh:8]: Which DIR? `//...` means "everything". Since there are no experimental dir here then it is good enough.

Comment (xkszltl) [reply scripts/codebase_utils.sh:8]: CRIS_CODE_DIRS, at least set it to empty string to keep interface consistent.

Comment (chenhao94) [reply scripts/codebase_utils.sh:8]: That does not work. CRIS_CODE_DIRS will be parsed to bazel targets `//dirname/...`, but for here we need `//...` to build all, and `///...` does not work

Comment (xkszltl) [reply scripts/codebase_utils.sh:8]: Current format_all relies on it.

Comment (chenhao94) [reply scripts/codebase_utils.sh:8]: Ack

**Review by xkszltl — APPROVED** [review]

**Pull request merged**